src/main/java/life/qbic/util/ProgressBar.java

```diff
@@ -27,11 +30,35 @@ public ProgressBar(String fileName, long totalFileSize) {
     this.stepSize = (float) totalFileSize / (float) BARSIZE;
     this.nextProgressJump = this.stepSize;
     this.start = System.currentTimeMillis();
+    lastUpdated = start + updateInterval;
   }
 
   public void updateProgress(int addDownloadedSize) {
     this.downloadedSize += (long) addDownloadedSize;
-    checkForJump();
+    update();
+  }
+
+  /**
+   * Updates the progress bar if an update is applicable.
+   */
+  public void update() {
+    if (progressStepsChanged()) {
+      this.nextProgressJump += this.stepSize;
+      drawProgress();
+    }
+    // update periodically
+    if (lastUpdateToOld()) {
```

Comment:
Grammatically nit: To differs from too 🤓 
```suggestion
    if (lastUpdateTooOld()) {
```